public/app/features/teams/TeamSettings.tsx

```diff
@@ -21,13 +24,28 @@ export type Props = ConnectedProps<typeof connector> & OwnProps;
 
 export const TeamSettings: FC<Props> = ({ team, updateTeam }) => {
   const canWriteTeamSettings = contextSrv.hasPermissionInMetadata(AccessControlAction.ActionTeamsWrite, team);
+  const currentOrgId = contextSrv.user.orgId;
+
+  const [{ roleOptions }, setOrgId] = useRoleOptions(currentOrgId);
+  const [pendingRoles, setPendingRoles] = useState<Role[]>([]);
+
+  const canUpdateRoles =
+    contextSrv.hasPermission(AccessControlAction.ActionUserRolesAdd) &&
+    contextSrv.hasPermission(AccessControlAction.ActionUserRolesRemove);
+
+  useEffect(() => {
+    setOrgId(currentOrgId);
+  }, [currentOrgId, setOrgId]);
 
   return (
     <VerticalGroup>
-      <FieldSet label="Team settings">
+      <FieldSet label="Team details">
         <Form
           defaultValues={{ ...team }}
           onSubmit={(formTeam: Team) => {
+            if (contextSrv.licensedAccessControlEnabled() && canUpdateRoles) {
+              updateTeamRoles(pendingRoles, team.id);
```

Comment:
This seems to be a long running query, especially if there are a lot of roles selected, we need the `Update` button to reflect this loading state, otherwise it's not clear what's happening and users might click `Update` button multiple times, especially on slower connections. 

Also I've noticed that sometimes when I click `Update` after applying a new role, I get an error `"database is locked"` from backend. Making this onSubmit callback async and adding `await` to `updateTeamRoles` should fix it.

Reply:
Fixed, thanks!